share/spice/timer/timer.js

```diff
@@ -406,19 +440,31 @@ License: CC BY-NC 3.0 http://creativecommons.org/licenses/by-nc/3.0/
     });
 
     env.ddg_spice_timer = function(api_result) {
+        var timers = [],
+            timerInterval,
+            $addTimerBtn;
+
+        function addTimer(startingTime) {
+            var timer = new Timer(timers.length + 1, startingTime);
+            timer.$element.insertBefore($addTimerBtn.parent());
+            timers.push(timer);
+        }
         
         function onShow() {
```

Comment:
I think really we should set onShow to fire once be default. We've been adding logic to several IA's now to explicitly check if the event has fired and prevent any further calls.

Alternatively we should make a new `init()` or `build()` (like we have for Goodies) that only fires once because that's almost always what we want out of the `onShow()` event